src/GitVersion.Core/VersionCalculation/VersionCalculators/NextVersionCalculator.cs

```diff
@@ -264,7 +269,8 @@ IEnumerable<NextVersion> GetNextVersionsInternal()
                             foreach (var baseVersion in versionStrategy.GetBaseVersions(effectiveBranchConfiguration))
                             {
                                 log.Info(baseVersion.ToString());
-                                if (IncludeVersion(baseVersion, configuration.Ignore))
+                                if (versionStrategy is not FallbackVersionStrategy || IncludeVersion(baseVersion, configuration.Ignore))
```

Comment:
please revert the logic to exclude the FallbackVersionStrategy.The FallackVersionStrategy class is like every other version strategy class and should not handled explicit.

Reply:
You see, this is not a good design (and makes it hard to understand) if you apply a implicit knowledge about the fact that the base version fo `FallbackVersionStrategy` is already evaluated. You are skipping it because of performance reason.. We need to find a design on an abstracion level e.g. ICommit interface where we ensure this aspect. We need to come to a design where we have transparent calls.